docs/day-2-operations/Backups.md

```diff
@@ -0,0 +1,272 @@
+# 3scale Backup & Restore
+
+## Overview
+
+This section intends to provide an Operator of a 3scale installation the information they need to:
+
+* Setup the backup procedures for persistent data
+* Perform a restore from backup of the persistent data
+
+Such that, in the case of a failure, they can restore 3scale to correctly running operational state and continue operating.
+
+### Persistent Volumes
+In a 3scale deployment on OpenShift all persistent data is stored either in a storage service in the cluster (not currently used), a Persistent Volume (PV) provided to the cluster by the underlying infrastructure or a storage service external to the cluster (be that in the same Data Center or elsewhere).
+
+### Considerations
+The backup and restore procedures for persistent data are required to vary depending on the storage used, to ensure the backups and restores preserve data conistsency (e.g. that a partial write, or a partial transaction is not captured). i.e. It is not sufficient to backup the underlying Persistent Volumes for a database, but instead the databases backup mechanisms should be used.
+
+Also, some parts of the data are synchronized between different components. One copy is considered the "source of truth" for the data set, and the other a copy that is not modified locally, just synchronized from the "source of truth". In these cases, upon restore, the "source of truth" should be restored and then the copies in other components synchronized from it.
+
+## Data Sets
+This section goes into more detail on the different data sets in the different persistent stores, their purposes, the storage type used and if it is the "source of truth" or not.
+
+The full state of a 3scale deployment is stored across these services and their persistent volumes:
+
+* `system-mysql`:  Mysql database (`mysql-storage`)
+* `system-storage`: Volume for Files
+* `zync-database`: Postgres database for `zync` component. This uses "HostPath" as storage. If the pod is moved into another node the data is lost. That is not a problem because the data are sync jobs and does not need to be 100% persistent.
```

Comment:
When the job was not triggered yet and the database is lost, then the sync will not happen unless an update is made on system side.

So losing data may lead to lost symchronization. The way to recover would be for `system` to trigger the synchronization like for `backend` sync

Reply:
@hallelujah so you think we should make this storage persistent and move away from `emptyDir` storage?

Reply:
Well not necessarily. If system can replay/resync events then it is ok

Reply:
Not sure System will have the information to know all the syncs required via Zync?
If it does, then sure, I agree...but we need a way in system to force this resync.

On the other hand, having a backup of this DB, and worst case doing the twice, would not be a problem. 

When we have System using Postgres, what storage does it need?
Will need to be persisted and backedup, no?
Then it would make sense to just have Zync use the same Postgres as System, and use the same back-up/restore procedures. If a few syncs get repeated, no problem, no?

Reply:
If System uses Postgres, it needs to be persistent storage. So zync will benefit of it if it is using the same database.

In case of different databases, then the only way for System to re-trigger the sync is the ability to replay the events that we store (for a short amount of time) in `event_store_events` table.
But this *replay* feature is not done.

Reply:
@andrewdavidmackenzie @hallelujah, in this case, could we resolve this conversation?

Reply:
@hallelujah do we have an issue for this replay feature?